From a016bf4b8fb305bfa24cb57b17bfacfdc449a905 Mon Sep 17 00:00:00 2001
From: David Wittman <dwittman@gmail.com>
Date: Sat, 28 Jan 2017 05:03:37 +0000
Subject: [PATCH] Add eudyptula id attribute to rtl8187 driver

Signed-off-by: David Wittman <dwittman@gmail.com>
---
 drivers/net/wireless/rtl818x/rtl8187/dev.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/drivers/net/wireless/rtl818x/rtl8187/dev.c b/drivers/net/wireless/rtl818x/rtl8187/dev.c
index 629ad8c..e9357e1 100644
--- a/drivers/net/wireless/rtl818x/rtl8187/dev.c
+++ b/drivers/net/wireless/rtl818x/rtl8187/dev.c
@@ -43,6 +43,25 @@ MODULE_AUTHOR("Larry Finger <Larry.Finger@lwfinger.net>");
 MODULE_DESCRIPTION("RTL8187/RTL8187B USB wireless driver");
 MODULE_LICENSE("GPL");

+/* Eudyptula ID */
+static char *id = "5d658d788cc9";
+
+static ssize_t id_show(struct device *dev, struct device_attribute *attr,
+		       char *buf)
+{
+	return sprintf(buf, "%s\n", id);
+}
+
+static ssize_t id_store(struct device *dev, struct device_attribute *attr,
+			const char *buf, size_t count)
+{
+	if (count - 1 == strlen(id) && !strncmp(buf, id, strlen(id)))
+		return count;
+
+	return -EINVAL;
+}
+static const DEVICE_ATTR_RW(id);
+
 static struct usb_device_id rtl8187_table[] = {
 	/* Asus */
 	{USB_DEVICE(0x0b05, 0x171d), .driver_info = DEVICE_RTL8187},
@@ -1438,6 +1457,10 @@ static int rtl8187_probe(struct usb_interface *intf,
 	int err, i;
 	u8 mac_addr[ETH_ALEN];

+	err = device_create_file(&udev->dev, &dev_attr_id);
+	if (err)
+		return -ENOMEM;
+
 	dev = ieee80211_alloc_hw(sizeof(*priv), &rtl8187_ops);
 	if (!dev) {
 		printk(KERN_ERR "rtl8187: ieee80211 alloc failed\n");
@@ -1660,6 +1683,7 @@ static void rtl8187_disconnect(struct usb_interface *intf)
 	if (!dev)
 		return;

+	device_remove_file(&intf->dev, &dev_attr_id);
 #ifdef CONFIG_RTL8187_LEDS
 	rtl8187_leds_exit(dev);
 #endif
--
2.1.4

